flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -819,28 +818,22 @@ class ExprLowering {
     }
     if constexpr (TC == Fortran::common::TypeCategory::Character) {
       fir::SequenceType::Shape shape;
-      shape.push_back(con.LEN());
       shape.append(con.shape().begin(), con.shape().end());
       auto chTy =
-          converter.genType(Fortran::common::TypeCategory::Character, KIND);
+          fir::CharacterType::get(builder.getContext(), KIND, con.LEN());
```

Comment:
Once upon a time, we had the idea of going through the converter to build types instead of building them inline wherever it seemed pleasant to do so. We've wandered away from that, but this change reminded me.
(It may become useful to have a standard type builder for gnarly derived types, for example, rather than just instantiating them willy nilly.)

Reply:
Good point, here I agree we should keep using it. I modified  `converter.genrType(cat, kind)`  to accept an arrayref as suggested below so that I could keep using it here.

FYI, in CharacterExpr.cpp (so not here), I will keep the bare uses of `fir::CharacterType::...` because  it is not dealing with any front-end data structures. My initial idea is that it is implementing character operations at the fir level that could one day be promoted as fir op (fir.substring, fir.concat...) if needed. To make this transition as painless as possible, it avoids using tools that would be not be movable/easily replaceable in codegen.

Reply:
Sounds good. Single point of truth is better design and lower maintenance cost IMHO.